macros/utils/table_operations/delete_and_insert.sql

```diff
@@ -57,7 +57,16 @@
 {% macro spark__get_delete_and_insert_queries(relation, insert_relation, delete_relation, delete_column_key) %}
     {% set queries = [] %}
 
-    {% if delete_relation %}
+    {% if delete_relation and relation.is_delta %}
```

Comment:
Cool @theromsh !
A couple of notes:

- I think it will be a bit cleaner to do:
```
{% if delete_relation %}
   {% if relation.is_delta %}
      ...
   {% else %}
      ...
   {% endif %}
{% endif %}

- Please add a comment clarifying that DELETEs are not support in the `is_delta` mode

Thanks!
```

Reply:
I don't think nesting condition should be a good practice, because it's less readable in my opinion (you always have to memorize the previous conditions).
Here's an article that explains why nested ifs should be avoided 
https://lawyerdev.medium.com/i-never-write-nested-ifs-e4e91a5440ee

Reply:
It's not very critical, but I admit I don't fully agree with that article :)
Especially since reading `{% elif delete_relation %}` forces me to remember the previous condition, and that it actually means both `delete_relation` and `not relation.is_delta`.

But I'm fine with keeping it as-is, as it's overall pretty clear.
I can merge tomorrow probably.